accord-core/src/main/java/accord/messages/CallbackHolder.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accord.messages;
+
+import accord.coordinate.Timeout;
+import accord.local.CommandStore;
+import accord.local.Node;
+import accord.local.PreLoadContext;
+
+public class CallbackHolder<T extends Reply>
```

Comment:
open to better names...

Reply:
`SafeCallback`, `CallbackInvoker`, `IndirectCallback`, `CallbackContext`?

It might also be nice to make `CommandStore` extend `Executor` so that we can support a cheap `execute` and in this case we can also call this e.g. `CallbackWithExecutor` (but this would be a nice change, even if we don't like this name)

Reply:
I like `Context` a lot but `SafeCallback` is more consistent with all the other `Safe*` interfaces that imply the work is in-store...